python/paddle/v2/fluid/distribute_transpiler.py

```diff
@@ -421,23 +464,89 @@ def _append_pserver_ops(self, program, pserver_program, opt_op, endpoint):
 
     def _append_pserver_non_opt_ops(self, program, pserver_program, opt_op):
         # Append the ops for parameters that do not need to be optimized/updated
-        for _, var in opt_op.inputs.iteritems():
-            program.global_block().create_var(
-                name=var.name,
-                persistable=var.persistable,
-                dtype=var.dtype,
-                shape=var.shape)
-            pserver_program.global_block().create_var(
-                name=var.name,
-                persistable=var.persistable,
-                dtype=var.dtype,
-                shape=var.shape)
+        if opt_op.inputs:
+            for _, vars in opt_op.inputs.iteritems():
+                if not isinstance(vars, list):
+                    vars = [vars]
+                for var in vars:
+                    program.global_block().create_var(
+                        name=var.name,
+                        persistable=var.persistable,
+                        dtype=var.dtype,
+                        shape=var.shape)
+                    pserver_program.global_block().create_var(
```

Comment:
Can add a FIXME here, should only create var once.

Reply:
Done.